### DISCUSSION TITLE
Fixed Grid column width

### DISCUSSION THREAD

**Discussion by LucaCris** [vikramlearning/blazorbootstrap#220] [Q&A]
How can I set a fixed column width in a Grid? To solve this problem (white space after badge icon):

![Screenshot 2023-06-01 123257](https://github.com/vikramlearning/blazorbootstrap/assets/50824617/c99d14e4-db91-4005-bb61-613ac6ceda9d)

**Comment by gvreddy04** [comment]
@LucaCris Right now fixed length is not supported. This grid is based on the bootstrap table. If you want, you can put custom width(ex: width:10%;).

**Reply by LucaCris** [reply]
Where is the "entry point" to set column width using style (good for me)?

**Comment by gvreddy04** [comment]
@LucaCris Can you share the sampe code. I'll check the alternate options.

**Comment by gvreddy04** [comment]
@LucaCris  No plans to fix this at this moment. This requires a custom code.

**Comment by thatplatypus** [comment]
I wanted to do this exact thing with some icons in grids last week. On the `GridColumn` component, I ended up using the `ColumnClass` property, passing it `"@(_ => "grid-col-icon")"`. I defined `grid-col-icon` somewhere for a specific width and then just re-used that pattern everywhere I had a column that was just an icon. Worked great for me.

Note that when it's loading it uses different classes so the width won't apply until it loads and applies the function/css class.

**Comment by narendrag06** [comment]
Hi,
Here i am trying to set the width of the GridColumn with CSS ColumnClass="@(_ => "grid-message-width" )". still its not working as expected. Can any one help is there anything i missed.
CSS :
.grid-message-width {
    width:10%;
}
